### PR TITLE
improved the font dialog

### PR DESCRIPTION
**Pull request by oltolm**
The font dialog was slow. I replaced the two combo boxes with native font picker dialogs. I like the new dialog more and it's faster.

I also reformatted the HTML code, but it's a minor detail.

### CODE REVIEW

**Comment by rzvncj** [comment]
Thanks!

At first glance, we need to run `clang-format -i` on all modified files (chmfontdialog.h in particular is now no longer properly indented).

But the deeper, philosophical issue, is that the original dialog had a single font size selector. With these changes, the font size can be selected for each of type of font (regular and fixed-size), with the setting for the fixed-size font being ignored.

Since [wxHtmlWindow::SetFonts()](https://docs.wxwidgets.org/3.0/classwx_html_window.html#a31b77979403242da3ccd40bea9b29b87) takes a single size parameter, that applies to both regular and fixed size fonts, we can't set their size separately even if we wanted to. But having the user modifying settings with no effect is also bad.

Is it possible to have setting the font size for either type of font also affect the other with the built-in font-dialog control? If not, I'm afraid I prefer the current code - the current font dialog being slower to start than the default wxWidgets one is not really a problem, I expect people will just set their preferred fonts once or twice and forget about it.

**Comment by oltolm** [comment]
I have reformatted `chmfontdialog.h` and changed the code to sync the file pickers. I personally would not sync them, but it's your choice.

**Comment by rzvncj** [comment]
> I have reformatted `chmfontdialog.h` and changed the code to sync the file pickers. I personally would not sync them, but it's your choice.

Why not? Is there another way to solve the problem I've raised?
Do you not agree that having a setting that doesn't do what it implies it would do based on the dialog preview is not ideal?

I'm not trying to impose something, just starting a conversation.

**Comment by oltolm** [comment]
I thought some more about it. Each solution has pros and cons. How often does a user change the fonts anyways? This one good enough.

**Comment by rzvncj** [comment]
> I thought some more about it. Each solution has pros and cons. How often does a user change the fonts anyways? This one good enough.

Do you mean you want to drop the PR?

**Comment by oltolm** [comment]
No, I mean this PR is good enough. I don't want to drop it.

**Comment by rzvncj** [comment]
> No, I mean this PR is good enough. I don't want to drop it.

Got it. I'll review tomorrow, unfortunately I'm swamped with other things tonight.

File: src/chmfontdialog.cpp [hunk 1]
```diff
@@ -111,8 +115,17 @@ void CHMFontDialog::UpdatePreview()
...
 {
+    if (event.GetId() == ID_FixedFont) {
+        wxFont selectedFont = _normalFControl->GetSelectedFont();
+        selectedFont.SetPointSize(event.GetFont().GetPointSize());
+        _normalFControl->SetSelectedFont(selectedFont);
+    } else if (event.GetId() == ID_NormalFont) {
+        wxFont selectedFont = _fixedFControl->GetSelectedFont();
+        selectedFont.SetPointSize(event.GetFont().GetPointSize());
+        _fixedFControl->SetSelectedFont(selectedFont);
+    }
```
Comment (rzvncj) [inline src/chmfontdialog.cpp]: But I think we don't need the IDs at all, and the more vebose `if ()` code can simply be replaced by:
```c++
_normalFont.SetPointSize(event.GetFont().GetPointSize());
_fixedFont.SetPointSize(event.GetFont().GetPointSize());
```
as in:
```diff
diff --git a/src/chmfontdialog.cpp b/src/chmfontdialog.cpp
index 5e57efc..ee4fe80 100644
--- a/src/chmfontdialog.cpp
+++ b/src/chmfontdialog.cpp
@@ -117,6 +117,7 @@ void CHMFontDialog::UpdatePreview()

 void CHMFontDialog::OnUpdate(wxFontPickerEvent& event)
 {
+    /*
     if (event.GetId() == ID_FixedFont) {
         wxFont selectedFont = _normalFControl->GetSelectedFont();
         selectedFont.SetPointSize(event.GetFont().GetPointSize());
@@ -126,9 +127,14 @@ void CHMFontDialog::OnUpdate(wxFontPickerEvent& event)
         selectedFont.SetPointSize(event.GetFont().GetPointSize());
         _fixedFControl->SetSelectedFont(selectedFont);
     }
+    */
+
     _normalFont = _normalFControl->GetSelectedFont();
     _fixedFont  = _fixedFControl->GetSelectedFont();

+    _normalFont.SetPointSize(event.GetFont().GetPointSize());
+    _fixedFont.SetPointSize(event.GetFont().GetPointSize());
+
     UpdatePreview();
 }
```
Am I missing something?

**Review by rzvncj — CHANGES_REQUESTED** [review]

File: src/chmfontdialog.cpp (hunk 1 above)
Comment (oltolm) [reply src/chmfontdialog.cpp]: That works too. I changed it.

**Comment by rzvncj** [comment]
Thanks!

**Pull request merged**